src/public/modules/forms/base/directives/submit-form.directive.js

```diff
@@ -429,6 +429,10 @@ function submitFormDirective(
           }
         })
       }
+      // Variable to indicate if there are prefill fields
+      scope.hasPrefill = scope.form.form_fields.some((field) => {
+        return field.allowPrefill
+      })
```

Comment:
can I clarify on what's the expected behavior if prefill is used in conjunction with logic? would that necessitate the use of `scope.$watch`?

Reply:
thanks, the warning message should show only when the field is visible. adjusted accordingly.